uber/site_sections/registration.py

```diff
@@ -306,48 +298,79 @@ def undo_sale(self, session, id):
         session.delete(session.sale(id))
         return 'Sale deleted'
 
+    def check_in_form(self, session, id):
+        attendee = session.attendee(id)
+        return {
+            'attendee': attendee,
+            'groups': [
+                (group.id, (group.name if len(group.name) < 30 else '{}...'.format(group.name[:27], '...'))
+                         + (' ({})'.format(group.leader.full_name) if group.leader else ''))
+                for group in session.query(Group)
+                                    .options(joinedload(Group.leader))
+                                    .filter(Group.status != c.WAITLISTED,
+                                            Group.id.in_(
+                                                session.query(Attendee.group_id)
+                                                       .filter(Attendee.group_id != None, Attendee.first_name == '')
+                                                       .distinct().subquery()))
+                                    .order_by(Group.name)
+            ] if attendee.paid == c.PAID_BY_GROUP and not attendee.group_id else []
+        }
+
     @ajax
-    def check_in(self, session, id, age_group, group, badge_num=None, message=''):
+    def check_in(self, session, id, age_group=None, birthdate=None, group=None, badge_num=None, message=''):
```

Comment:
Beyond that, I did some refactoring of this function.  There were a few minor problems I noticed which don't directly affect MAGFest but could affect events with certain configurations.  For example, we were only doing group assignments if numbered badges were turned on.

I broke everything out into its own "top level" check, and added some new error handling cases.  Because this changed the indentation of most of this file, it's kind of hard to read, but there aren't actually many changes here.